File: 1013-fibonacci-number/1013-fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        if(n==0){
            return 0;
        }
        int z = 0;
        int c = 0;
        int k = 1;
        vector<int> arr(n); 
        arr.push_back(c);
        arr.push_back(k);
        
        for (int i = 2; i < n; i++) {
            int r = c + k;
            c = k;
            k = r;
            arr.push_back(r);
        }
        z = arr[arr.size()-1] + arr[arr.size()-2];
        return z;
    }
};

File: 1013-fibonacci-number/1013-fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        if(n==0):
            return 0
        c=0
        k=1
        z=0
        l=[]
        l.append(c)
        l.append(k)
        for i in range(2,n):
            z=c+k
            c=k
            k=z
            l.append(z)
        k=l[len(l)-1]+l[len(l)-2]
        return k
        